internal/controller/stackrunjob_controller.go

```diff
@@ -72,15 +74,24 @@ func (r *StackRunJobReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// Update step statuses, i.e., when stack run was successful or failed.
 	for _, step := range stackRun.Steps {
 		if update := r.getStepStatusUpdate(stackRun.Status, step.Status); update != nil {
-			err := r.ConsoleClient.UpdateStackRunStep(step.ID, console.RunStepAttributes{Status: *update})
-			return ctrl.Result{}, err
+			if err := r.ConsoleClient.UpdateStackRunStep(step.ID, console.RunStepAttributes{Status: *update}); err != nil {
+				return ctrl.Result{}, err
+			}
 		}
 	}
 
 	// Exit if stack run is not in running state (run status already updated),
 	// or if the job is still running (harness controls run status).
 	if stackRun.Status != console.StackStatusRunning || job.Status.CompletionTime.IsZero() {
-		return ctrl.Result{}, nil
+		if isActiveJobTimout(stackRun.Status, job) {
```

Comment:
I think we should also manually delete the job here, or perhaps kill its pod.  I'm not sure if the ttl on the job will pick up

Reply:
done